src/Controller.php

```diff
@@ -0,0 +1,50 @@
+<?php
+declare(strict_types=1);
+
+namespace Magento\Upward;
+
+class Controller
+{
+    /** @var \Zend\Http\PhpEnvironment\Request */
+    private $request;
+
+    /** @var Context */
+    private $context;
+
+    /** @var Definition */
+    private $definition;
+
+    /** @var DefinitionTraverser */
+    private $definitionTraverser;
+
+    public function __construct(
+        \Zend\Http\PhpEnvironment\Request $request
+    ) {
+        $this->request = $request;
+        $this->context = Context::fromRequest($request);
+        // this value will need to be sourced from config?
+        $this->definition = Definition::fromYamlFile("pwa/upward-config-sample.yml");
+        $this->definitionTraverser = new DefinitionTraverser($this->definition, $this->context);
+    }
+
+    /**
+     * Executes request and returns response
+     *
+     * @return \Zend\Http\Response
+     */
+    public function run(): \Zend\Http\Response
```

Comment:
My thought was this would be a good use case for `__invoke()` but I'm not absolutely stuck on that idea.

Reply:
Made this change assuming this would be the only public method. I'm not a fan of calling `$controller()` in `bin/upward`, if you have a better naming convention.

Reply:
🤷‍♂️ Naming stuff is one of the two hard things in programming.

Calling this class a "Controller" was the closest concept I could think of for it's purpose, but I welcome any better ideas.